bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -5938,6 +6043,205 @@ private static void closeStatement(Statement statement) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	void setIngestStatus(int ingestJobId, IngestJobStatusType status) throws TskCoreException, TskDataException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		ResultSet resultSet = null;
+		try {
+			Statement statement = connection.createStatement();
+			resultSet = statement.executeQuery("SELECT ingest_job_id FROM ingest_jobs WHERE ingest_job_id=" + ingestJobId);
+			if (resultSet.next()) {
+				statement.executeUpdate("UPDATE ingest_jobs SET status_id=" + status.ordinal()+ " WHERE ingest_job_id=" + ingestJobId + ";");
+			} else {
+				throw new TskDataException("Given ingest job was not found in database.");
+			}
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
+	/**
+	 * Adds the given ingest job to the database
+	 *
+	 * @param dataSource    The datasource the ingest job is being run on
+	 * @param hostName      The name of the host
+	 * @param ingestModules The ingest modules being run during the ingest job
+	 * @param jobStart      The time the job started
+	 *
+	 * @return An information object representing the ingest job added to the
+	 *         database.
+	 *
+	 * @throws TskCoreException If adding the job to the database fails.
+	 */
+	public final IngestJobInfo addIngestJob(Content dataSource, String hostName, List<IngestModuleInfo> ingestModules, long jobStart) throws TskCoreException {
+		CaseDbConnection connection = connections.getConnection();
+		acquireSharedLock();
+		ResultSet resultSet = null;
+		Statement statement = null;
+		try {
+			statement = connection.createStatement();
+			resultSet = statement.executeQuery("SELECT MAX(ingest_job_id) FROM ingest_jobs");
+			int id = 0;
+			if (resultSet.next()) {
+				//We want 1 + the maximum job id.
+				id = resultSet.getInt(1) + 1;
+			}
+			statement.executeUpdate("INSERT INTO ingest_jobs (ingest_job_id, data_src_id, host_name, start_date, end_date, status_id, settings_dir) "
+					+ "VALUES (" + id + ", " + dataSource.getId() + ", '" + hostName + "', " + jobStart + ", 0, 0, '');");
+			for (int i = 0; i < ingestModules.size(); i++) {
+				IngestModuleInfo ingestModule = ingestModules.get(i);
+				statement.executeUpdate("INSERT INTO ingest_job_modules (ingest_job_id, ingest_module_id, position) "
+						+ "VALUES (" + id + ", " + ingestModule.getIngestModuleId() + ", " + i + ");");
+			}
+			return new IngestJobInfo(id, dataSource.getId(), hostName, new Date(jobStart), "", ingestModules, this);
+		} catch (SQLException ex) {
+			throw new TskCoreException("Error updating the end date.", ex);
+		} finally {
+			closeResultSet(resultSet);
+			connection.close();
+			releaseSharedLock();
+		}
+	}
+
```

Comment:
The addIngestJobInfo and addIngestModuleInfo methods should not do the SELECT to get the PKs for the INSERTS. The PKs must be auto-generated. Use prepared statements if needed, e.g., 

			// Insert a row for the virtual directory into the tsk_objects table.
			// INSERT INTO tsk_objects (par_obj_id, type) VALUES (?, ?)
			CaseDbConnection connection = transaction.getConnection();
			PreparedStatement statement = connection.getPreparedStatement(PREPARED_STATEMENT.INSERT_OBJECT, Statement.RETURN_GENERATED_KEYS);
			statement.clearParameters();
			if (parentId != 0) {
				statement.setLong(1, parentId);
			} else {
				statement.setNull(1, java.sql.Types.BIGINT);
			}
			statement.setInt(2, TskData.ObjectType.ABSTRACTFILE.getObjectType());
			connection.executeUpdate(statement);
			resultSet = statement.getGeneratedKeys();
			resultSet.next();
			long newObjId = resultSet.getLong(1);